lib/core/base/audit.js

```diff
@@ -40,8 +40,191 @@ function Audit(audit) {
 
 	this.defaultConfig = audit;
 	this._init();
+
+	// A copy of the "default" locale. This will be set if the user
+	// provides a new locale to `axe.configure()` and used to undo
+	// changes in `axe.reset()`.
+	this._previousLocale = null;
 }
 
+/**
+ * Build and set the previous locale. Will noop if a previous
+ * locale was already set, as we want the ability to "reset"
+ * to the default ("first") configuration.
+ */
+
+Audit.prototype._setPreviousLocale = function() {
+	if (this._previousLocale) {
+		return;
+	}
+
+	const locale = {
+		checks: {},
+		rules: {}
+	};
+
+	// XXX: unable to use `for-of` here, as doing so would
+	// require us to polyfill `Symbol`.
+	const checkIDs = Object.keys(this.data.checks);
+	for (let i = 0; i < checkIDs.length; i++) {
+		const id = checkIDs[i];
+		const check = this.data.checks[id];
+		const { pass, fail, incomplete } = check.messages;
+		locale.checks[id] = {
+			pass,
+			fail,
+			incomplete
+		};
+	}
+
+	const ruleIDs = Object.keys(this.data.rules);
+	for (let i = 0; i < ruleIDs.length; i++) {
+		const id = ruleIDs[i];
+		const rule = this.data.rules[id];
+		const { description, help } = rule;
+		locale.rules[id] = { description, help };
+	}
+
+	this._previousLocale = locale;
+};
+
+/**
+ * Reset the locale to the "default".
+ */
+
+Audit.prototype._resetLocale = function() {
+	// If no previous locale exists, then we haven't applied one
+	// and the default is still in-tact.
+	const previousLocale = this._previousLocale;
+	if (!previousLocale) {
+		return;
+	}
+
+	// Apply the previous locale
+	this.applyLocale(previousLocale);
+};
+
+/**
+ * Merge two check locales (a, b), favoring `b`.
+ *
+ * Both locale `a` and the returned shape resemble:
+ *
+ *    {
+ *      impact: string,
+ *      messages: {
+ *        pass: string | function,
+ *        fail: string | function,
+ *        incomplete: string | {
+ *          [key: string]: string | function
+ *        }
+ *      }
+ *    }
+ *
+ * Locale `b` follows the `axe.CheckLocale` shape and resembles:
+ *
+ *    {
+ *      pass: string,
+ *      fail: string,
+ *      incomplete: string | { [key: string]: string }
+ *    }
+ */
+
+const mergeCheckLocale = (a, b) => {
+	let { pass, fail } = b;
+	// If the message(s) are Strings, they have not yet been run
+	// thru doT (which will return a Function).
+	if (typeof pass === 'string') {
+		pass = axe.imports.doT.compile(pass);
+	}
+	if (typeof fail === 'string') {
+		fail = axe.imports.doT.compile(fail);
+	}
+	return {
+		impact: a.impact,
```

Comment:
What if they want to override the impact?

Reply:
This is not currently supported with the existing localization setup. Adding it now falls out of scope of the task I was given.

If it's something we know we want, we can add support after this PR lands.

Reply:
+1 to Stephen's comment. You can change impact by doing this:

```js
axe.configure({
  checks: [{
    id:'foo',
    metadata: { impact: 'minor' }
  }]
});
```

I think that instead of copying over `impact`, we should just spread the rest of the props in (`...a`). That way we can't accidentally forget to copy stuff over if we add stuff to metadata in the future.

Reply:
👍 I'll spread `a` here.